### ISSUE TITLE
[share_plus] PNG image being exported as JPG

### ISSUE THREAD

**Issue by valdineifer** [fluttercommunity/plus_plugins#859]
Hello, I didn't want to create an issue to ask for help, but I didn't get the solution for my problem anywhere. I tried an national Flutter group and [Stack Overflow (where I didn't get any answer)](https://stackoverflow.com/questions/72153339/how-to-share-files-with-a-different-extension-in-flutter). This is my code:
```
    final appDir = await syspaths.getApplicationDocumentsDirectory();
    final fileName = path.basenameWithoutExtension(imageXFile.path);

    img.Image? image = img.decodeImage(File(imageXFile.path).readAsBytesSync());

    final imageFile = File('${appDir.path}/$fileName.png');
    imageFile.writeAsBytesSync(img.encodePng(image));

    setState(() {
      _items.add(new Item(fileName, imageFile));
    });
```

I execute the sharing with `Share.shareFiles([fileInstance.path], mimeTypes: ['image/png'])`, but the image is exported with a name based on actual time and as JPG, like `IMG_20220510_143539_398.jpg`.

I don't know if I'm doing something wrong (I believe that, that's why I didn't think it's a bug), but I hope someone find out what is going on with this code.

**Comment by valdineifer** [comment]
Hello again.
**TL;DR**: My issue isn't related to the `share_plus` package. I still trying to find a solution. I will continue to update the Stack Overflow post.

I went to the Kotlin code to check if the type and filename still are equal to `fileInstance.path`. It is.
So it appears that Telegram (my target app) send the image as jpg ang generate the filename. When I tried to send as attach with Outlook, it appears as png.
So, sorry about the issue created.
